src/Compilers/Core/Portable/Symbols/ISymbolExtensions.cs

```diff
@@ -116,5 +118,86 @@ internal static bool IsInSource(this ISymbol symbol)
 
             return false;
         }
+
+        // Keep consistent with TypeSymbolExtensions.IsWellKnownTypeLock.
+        internal static bool IsLockType(this ITypeSymbol type)
+            => type is INamedTypeSymbol
```

Comment:
>=> [](http://example.com/codeflow?start=12&length=2)

Consider using regular block body. Given that the expression spans more than 10 lines of code, using an expression body doesn't provide any advantages. It actually negatively affects readability, in my opinion.